callstats-java-sdk/src/main/java/io/callstats/sdk/internal/CallStatsAuthenticator.java

```diff
@@ -184,7 +192,8 @@ public void run() {
 	 * @param httpClient the http client
 	 */
 	private void scheduleAuthentication(final int appId, final String bridgeId, final CallStatsHttpClient httpClient) {
-		scheduler.schedule(new Runnable() {
+		cancelScheduledAuthentication();
+		scheduledFuture = scheduler.schedule(new Runnable() {
```

Comment:
There's race condition.
T1 calls scheduleAuthentication
T2 calls scheduleAuthentication
T1 calls cancelScheduledAuthentication and it finishes (may remove something)
T2 calls cancelScheduledAuthentication and it finishes (won't remove anything)
T1 schedules authentication
T2 schedules authentication

Now there are 2 authentication attempts scheduled.